synthtool/transforms.py

```diff
@@ -60,9 +61,23 @@ def _expand_paths(paths: ListOfPathsOrStrs, root: PathOrStr = None) -> Iterable[
             )
 
 
+def _file_is_gitignored(path: Path) -> bool:
+    """Checks if a file is ignored by git"""
+    try:
+        subprocess.run(["git", "check-ignore", "-q", str(path.resolve())], stderr=subprocess.PIPE)
```

Comment:
oof, forking for every file is going to cause significant slowdown for synthtool. Can we do this a different way? Do we really need to avoid gitignored files? (I mean, they're ignored anyway).

Reply:
`git check-ignore` can take a list of files on standard input:  it only outputs those which are ignored. See the [manpage](https://git-scm.com/docs/git-check-ignore).

Reply:
That's useful, but I'd still like to understand why we care about synthtool touching ignored files?

Reply:
Lol I realize now it was [my suggestion](https://github.com/googleapis/synthtool/issues/67#issuecomment-420813577). I rescind it. Let's do not that for now.